tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11106,6 +11108,7 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC,
 
       switch (pAttr->getKind()) {
       case AttributeList::AT_HLSLLinear:
+      case AttributeList::AT_HLSLCenter:
```

Comment:
Track center independently from pLinear (use pCenter).  Verify/warn later if used with the other linear modifiers that conflict (centroid, sample).  Use with noperspective is ok.